File: Algorithms/src/2144. Minimum Cost of Buying Candies With Discount/Code/solution.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minimumCost = function (cost) {
  // Sort from highest cost to lowest cost
  cost.sort((a, b) => b - a);

  let total = 0;

  // Skip every third candy
  for (let i = 0; i < cost.length; i++) {
    if (i % 3 === 2) continue;

    total += cost[i];
  }

  return total;
};

File: Algorithms/src/2144. Minimum Cost of Buying Candies With Discount/Code/solution.go
func minimumCost(cost []int) int {
    // Sort candies from highest cost to lowest cost
    sort.Slice(cost, func(i, j int) bool {
        return cost[i] > cost[j]
    })

    total := 0

    // Skip every third candy
    for i := 0; i < len(cost); i++ {
        if i%3 == 2 {
            continue
        }

        total += cost[i]
    }

    return total
}